src/v2/normalized_packages.py

```diff
@@ -53,3 +57,65 @@ def dependency_graph(self) -> Dict[Package, Set[Package]]:
     def ecosystem(self):
         """Ecosystem value."""
         return self._ecosystem
+
+
+class GoNormalizedPackages(NormalizedPackages):
+    """Duplicate free list of GoNormalised Packages."""
+
+    def __init__(self, packages: List[Package], ecosystem: Ecosystem):
+        """Create NormalizedPackages by removing all duplicates from packages."""
+        self._packages = packages
+        self._ecosystem = ecosystem
```

Comment:
Base class fields should be initialized using Base class constructor. Accessing private members of base class violates encapsulation principle.

Reply:
`_` (Pre Single Underscore) in Python is for Protected Members. (This is what we are using in our codebase)
`__` (Pre Double Underscore) in Python is for Private Members. (Rarely used in our codebase)

Hence Protected Members can be accessed from Child Class. Should not be an issue. 
https://www.geeksforgeeks.org/access-modifiers-in-python-public-private-and-protected/